testgrid/cmd/configurator/prow.go

```diff
@@ -129,7 +131,16 @@ func applySingleProwjobAnnotations(c *Config, pc *prowConfig.Config, j prowConfi
 			dashboardName = strings.TrimSpace(dashboardName)
 			d := c.config.FindDashboard(dashboardName)
 			if d == nil {
-				return fmt.Errorf("couldn't find dashboard %q for job %q", dashboardName, j.Name)
+				if !generateYAML {
```

Comment:
I think we should insist on dashboards being defined in the same set of files as the things that go on them, and not let this pass without comment (and then potentially fail on our side, which will be more painful).

Reply:
I think I'd rather keep the YAML config as the authoritative source of dashboards for now; it lets us keep the typo protection and lets us keep blending config and annotations.

I'm interested in additional controls on what's defined where, especially if it keeps things easy to understand, but I think that's beyond this PR.

Reply:
I agree, but this code seems to have the opposite effect - if a dashboard doesn't exist, it will now just silently invent it and assume that's okay.

I'm not sure how you're envisioning running this, but it must involve some amount of yaml in the other repo to define their dashboards and whatnot too?

Reply:
For clarity, I very recently pushed a revision that removes this particular change; now it throws an error if it sees a dashboard that isn't in the YAML like it used to.

I originally envisioned the program simply taking a series of prow jobs with annotations and, assuming they're perfectly accurate, constructing a yaml configuration file. In the [first draft](https://github.com/kubernetes/test-infra/pull/13510), I had to specifically specify a GCS bucket and a configuration file for all those defaults.

On the first pass through of this, I thought it would be neat to have it generate dashboards as well, so you only need a prow config and also a testgrid config that contains some defaults.

After this conversation, I _now_ think that's way more error-prone than is worthwhile, and think it's best with a prow config and also a testgrid config that contains defaults and dashboards.